onnxruntime/core/providers/webnn/builders/impl/max_min_op_builder.cc

```diff
@@ -0,0 +1,100 @@
+
+namespace onnxruntime {
+namespace webnn {
+
+class MaxMinOpBuilder : public BaseOpBuilder {
+  // Add operator related.
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override ORT_MUST_USE_RESULT;
+
+  // Operator support related.
+  bool IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                         WebnnDeviceType /* device_type */, const logging::Logger& logger) const override;
+};
+
+// Add operator related.
+
+Status MaxMinOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
+                                                 const Node& node,
+                                                 const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  const auto& op_type = node.OpType();
+
+  emscripten::val input0 = model_builder.GetOperand(input_defs[0]->Name());
+  std::vector<int64_t> input_shape;
+  ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get shape");
+
+  const size_t input_count = input_defs.size();
+  ORT_RETURN_IF(input_count < 1, op_type, "has no inputs");
+
+  emscripten::val output = emscripten::val::object();
+
+  if (input_count == 1) {
+    output = input0;
+  } else {
+    ORT_RETURN_IF_NOT(op_type == "Max" || op_type == "Min", "MaxMinOpBuilder, unknown op: ", op_type);
+    std::string webnn_op_name = op_type == "Max" ? "max" : "min";
+
+    emscripten::val input1 = model_builder.GetOperand(input_defs[1]->Name());
+    output = model_builder.GetBuilder().call<emscripten::val>(webnn_op_name.c_str(), input0, input1);
+
+    for (size_t input_index = 2; input_index < input_count; ++input_index) {
+      emscripten::val next_input = model_builder.GetOperand(input_defs[input_index]->Name());
+      output = model_builder.GetBuilder().call<emscripten::val>(webnn_op_name.c_str(), output, next_input);
+    }
+  }
+
+  model_builder.AddOperand(node.OutputDefs()[0]->Name(), std::move(output));
+  return Status::OK();
+}
+
+// Operator support related.
+bool MaxMinOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& /* initializers */,
+                                           const Node& node,
+                                           WebnnDeviceType /* device_type */,
+                                           const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  const auto& op_type = node.OpType();
+
+  std::vector<int64_t> input_shape;
+  if (!GetShape(*input_defs[0], input_shape, logger))
+    return false;
+
+  if (input_defs.size() < 1) {
+    LOGS(logger, VERBOSE) << op_type << " requires at least one input (data)";
+    return false;
+  }
+
+  return true;
+}
+
+void CreateMaxMinOpBuilder(const std::string& op_type, OpBuilderRegistrations& op_registrations) {
+  if (op_registrations.op_builder_map.find(op_type) != op_registrations.op_builder_map.cend())
```

Comment:
```suggestion
  if (op_registrations.op_builder_map.count(op_type) > 0)
```

A little shorter. Using `contains(op_type)` would be even better, but that's C++20 only.